test/e2e/utils/cnpspecbuilder.go

```diff
@@ -218,6 +218,27 @@ func (b *ClusterNetworkPolicySpecBuilder) AddEgress(protoc v1.Protocol,
 	return b
 }
 
+func (b *ClusterNetworkPolicySpecBuilder) AddNodeSelectorRule(nodeSelector *metav1.LabelSelector, protoc v1.Protocol, port *int32, name string,
+	ruleAppliedToSpecs []ACNPAppliedToSpec, action crdv1alpha1.RuleAction) *ClusterNetworkPolicySpecBuilder {
+	var appliedTos []crdv1alpha1.NetworkPolicyPeer
+	for _, at := range ruleAppliedToSpecs {
+		appliedTos = append(appliedTos, b.GetAppliedToPeer(at.PodSelector, at.NSSelector, at.PodSelectorMatchExp, at.NSSelectorMatchExp, at.Group))
+	}
+	policyPeer := []crdv1alpha1.NetworkPolicyPeer{{NodeSelector: nodeSelector}}
+
+	newRule := crdv1alpha1.Rule{
+		To: policyPeer,
+		Ports: []crdv1alpha1.NetworkPolicyPort{
+			{Protocol: &protoc, Port: &intstr.IntOrString{IntVal: *port}},
+		},
+		Action:    &action,
+		Name:      name,
+		AppliedTo: appliedTos,
+	}
+	b.Spec.Egress = append(b.Spec.Egress, newRule)
```

Comment:
Is nodeSelector only supported for egress?

Reply:
Thanks for reviewing. Both egress and ingress rules, and will add more e2e test cases.